### DISCUSSION TITLE
Content Security Policy: Couldn’t process unknown directive ‘script-src-attr’ with Firefox

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#10928] [Archive]
Using Firefox (95.0.2), I could not get rid of 
`Content Security Policy: Couldn’t process unknown directive ‘script-src-attr’` 
with any of the proposed solutions in #10833
Edge supports `script-src-attr` but [not Firefox](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) where it is still a Firefox [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1529337)

Is there a Directus setting that adds meta to all pages? 
like 
`<meta http-equiv="Content-Security-Policy" content="default-src *; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' 'unsafe-eval' http://localhost/">`
as I read that this would help meanwhile the Firefox bug is corrected

Also, the Fifefox console shows
`Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”).`
on the creation page of a simple collection
What can i do?

**Comment by azrikahar** [comment]
> Edge supports script-src-attr but not Firefox where it is still a Firefox bug

Thanks for pointing this bug with Firefox!

You can try adding the following to remove `script-src-attr`:

```shell
CONTENT_SECURITY_POLICY_DIRECTIVES__SCRIPT_SRC_ATTR="null"
```

> Is there a Directus setting that adds meta to all pages?

It's due to [Helmet](https://helmetjs.github.io/), which in turn added Content Security Policy as one of it's security measures. They are sent as response headers rather than a meta tag in this case.

**Comment by RafeSafari** [comment]
> It's due to [Helmet](https://helmetjs.github.io/), which in turn added Content Security Policy as one of it's security measures. They are sent as response headers rather than a meta tag in this case.

azrikahar is right.
I used this code to remove that header:
```
app.use(
    helmet.contentSecurityPolicy({
        directives: {
            'script-src-attr': null
        }
    })
);
```

**Reply by notflip** [reply]
Where did you add that code? Are you using docker to run Directus?

**Comment by mushu999** [comment]
How would this be done in a web.config website under IIS in Windows please?
